include/seqan3/core/detail/empty_type.hpp

```diff
@@ -0,0 +1,25 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::empty_type.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+//!\brief An empty class type used in meta programming.
+//!\ingroup core
+struct empty_type
```

Comment:
I think we have this on other places, too, maybe move to core?

Reply:
Oh, it is in code already. Ok. One place that you need to change is inherited_iterator_base, because that exactly also defines this type (with the same name).

Reply:
It was moved from there

Reply:
Hm, I didn't see it being removed there, but maybe it was.